Add initial v1 REST API Documentation (unpublished)

juliawu opened the pull request:
This change adds a folder of v1 REST API documentation pages to the docsite, but unpublished. The intention is to add documentation to the folder incrementally in subsequent pull requests, so that we can release at once when all pages are ready and approved.

This starter set includes:
* a dedicated v1 REST folder
* landing page for said folder
* v1 REST API getting started guide
* /v1/observations/point + bulk version
* /v1/observations/series + bulk version

![Screen Shot 2022-07-27 at 3 30 16 PM](https://user-images.githubusercontent.com/4034366/181383632-bbc2f8ee-155a-42bc-aad5-3de0a66ff3db.png)

juliawu commented on _includes/sidebar.html:
```diff
@@ -38,9 +38,11 @@
-                      <li class="{% if page.url == grand_child.url %}active{% endif %}">
-                        <a href="{{ grand_child.url }}" class="{% if page.url == grand_child.url %} active{% endif %}">{{ grand_child.title }}</a>
-                      </li>
+                      {%- unless grand_child.nav_exclude -%}
```
This prevents the links to simple and bulk versions of the same API becoming duplicate entries in the sidebar

shifucun commented on api/rest/v1/bulk_observations_series.md:
```diff
@@ -0,0 +1,341 @@
+ 
+# /v1/bulk/observations/series
+
+Retrieve all values for multiple variables and entities.
```
all values sounds a bit too general, maybe "all values" => "observation series"? 

Here, the tips below, and **_point.md all mention "single point" and "all values". Might make this more explicit by saying "observation at a date" vs "observation time series" throughout consistently.

shifucun commented on api/rest/v1/bulk_observations_series.md:
```diff
@@ -0,0 +1,341 @@
+ 
+<div markdown="span" class="alert alert-warning" role="alert" style="color:black; font-size: 0.8em">
+   <span class="material-icons md-16">info </span><b>See Also:</b><br />
+   To retrieve single points within a series, use [/v1/bulk/observations/point](/api/rest/v1/observations/point)<br />.
```
single point

shifucun commented on api/rest/v1/getting_started.md:
````diff
@@ -0,0 +1,161 @@
+```json
+{
+  "code": 3,
+  "message": "Missing required argument: stat_var",
````
nit: stat_var -> variable

shifucun commented on api/rest/v1/getting_started.md:
````diff
@@ -0,0 +1,161 @@
+}
+```
+
+If your request includes a bad argument, you will receive a 404 status code and an error message like the following:
````
In these cases I think the APIs don't throw error while empty result is  returned for that entry.

shifucun commented on api/rest/v1/index.md:
```diff
@@ -0,0 +1,88 @@
+# Data Commons REST API
+
+ 
+The Data Commons REST API is a [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) library that enables developers to programmatically access nodes in the Data Commons knowledge graph. This package allows users to explore the structure of the graph, integrate statistics from the graph into data analysis applications and much more.
```
access data

shifucun commented on api/rest/v1/index.md:
```diff
@@ -0,0 +1,88 @@
+ 
+| API | URI | Description |
+| --- | --- | ------------|
+| Get a triple | /v1/triples | Get neighboring nodes and property labels of directed edges of a specific entity
```
Get triples

shifucun commented on api/rest/v1/observations_point.md:
```diff
@@ -0,0 +1,126 @@
+
+## Examples
+
+### Example 1: Get single value for given variable and entity
```
"Get latest observation for given variable and entity" to echo below "specific date"

juliawu commented:
Made edits in response to above comments:
- descriptions of APIs are now more specific: "single value" -> "specific observation at a set date" and "all values" -> "series of observations"
- Wording changes: "Get a triple" -> "Get triples" and "access nodes" -> "access data"
- Fixed double spacing issues
- reverted .gitignore

juliawu commented on api/rest/v1/bulk_observations_point.md:
```diff
@@ -0,0 +1,437 @@
+| entities <br /><required-tag>Required</required-tag> | Repeated string | DCIDs of the entities the variables describe. |
+| variables <br /><required-tag>Required</required-tag> | Repeated string | DCIDs of the variables to query observations for.|
+| date <br /> <optional-tag>Optional</optional-tag> | string | Datetime of  measurement of the value requested in ISO 8601 format. To see the dates available, look up the variable in the [Statistical Variable Explorer](https://datacommons.org/tools/statvar). If date is not provided, the latest available datapoint is returned.  |
+| all_facets <br /><optional-tag>Optional</optional-tag> | Boolean | Whether to return data from all facets available. If true, data from all facets available  will be returned. If false, only data  from the preferred facet will be returned. Defaults to false.
```
This first PR doesn't include the new concept page, will link in the next PR!

shifucun commented:
Thanks for all the update!
